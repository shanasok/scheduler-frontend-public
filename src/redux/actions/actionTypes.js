export const LOAD_COURSES = "LOAD_COURSES";
export const SET_CURRENT_LESSON = "SET_CURRENT_LESSON";
export const SET_CURRENT_PAGE = "SET_CURRENT_PAGE";
export const SET_CURRENT_LESSON_CONTENT = "SET_CURRENT_LESSON_CONTENT";
export const SET_EDIT_SLIDE_TRUE = "SET_EDIT_SLIDE_TRUE";
export const SET_EDIT_SLIDE_FALSE = "SET_EDIT_SLIDE_FALSE";
export const SET_PAGE_CONTENT_EDITABLE = "SET_PAGE_CONTENT_EDITABLE";