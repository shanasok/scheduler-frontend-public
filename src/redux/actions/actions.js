import { LOAD_COURSES } from "./actionTypes";

// Actions are payloads of information that send data from your application to your store.

export const loadCourses = courses => ({
    type: LOAD_COURSES,
    payload: {
        courses: courses
    }
});
