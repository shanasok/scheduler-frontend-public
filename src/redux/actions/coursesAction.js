import { LOAD_COURSES } from "./actionTypes";

// Actions are payloads of information that send data from your application to your store.

export function setCoursesJson(coursesJson) {
    return {
        type: LOAD_COURSES,
        payload: {
            courses: coursesJson
        }
    }
}
