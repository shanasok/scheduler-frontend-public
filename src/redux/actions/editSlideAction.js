import { SET_EDIT_SLIDE_TRUE, SET_EDIT_SLIDE_FALSE } from "./actionTypes"

// ActionlessonsAction.jss are payloads of information that send data from your application to your store.

export function setEditSlideTrue() {
    return {
        type: SET_EDIT_SLIDE_TRUE,
        payload: {
            setEditSlide: true
        }
    }
}

export function setEditSlideFalse() {
    return {
        type: SET_EDIT_SLIDE_FALSE,
        payload: {
            setEditSlide: false
        }
    }
}