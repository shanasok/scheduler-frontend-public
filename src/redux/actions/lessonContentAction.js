import { SET_CURRENT_LESSON_CONTENT } from "./actionTypes";

// ActionlessonsAction.jss are payloads of information that send data from your application to your store.

export function setCurrentLessonContent(currentLessonContent) {
    return {
        type: SET_CURRENT_LESSON_CONTENT,
        payload: {
            currentLessonContent: currentLessonContent
        }
    }
}