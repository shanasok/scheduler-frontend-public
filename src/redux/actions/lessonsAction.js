import { SET_CURRENT_LESSON } from "./actionTypes";

// ActionlessonsAction.jss are payloads of information that send data from your application to your store.

export function setCurrentLesson(currentLesson) {
    return {
        type: SET_CURRENT_LESSON,
        payload: {
            currentLesson: currentLesson
        }
    }
}
