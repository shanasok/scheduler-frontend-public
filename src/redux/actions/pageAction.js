import { SET_CURRENT_PAGE } from "./actionTypes";

// ActionlessonsAction.jss are payloads of information that send data from your application to your store.

export function setCurrentPage(currentPage) {
    return {
        type: SET_CURRENT_PAGE,
        payload: {
            currentPage: currentPage
        }
    }
}
