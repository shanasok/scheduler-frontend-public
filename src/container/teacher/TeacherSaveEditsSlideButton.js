import React from 'react';
import {bindActionCreators} from "redux";
import {connect} from "react-redux";
import { setEditSlideFalse } from '../../redux/actions/editSlideAction';

export function saveAndCloseSlideEditor(setEditSlideFalseFunction){
    setEditSlideFalseFunction();
}

class TeacherSaveEditsSlideButton extends React.Component {

    render() {
        return <button className="saveEditsButton curvedButton"
                       onClick={() => saveAndCloseSlideEditor(this.props.actions.setEditSlideFalse)}>
            Save Changes</button>
    }
}

const mapStateToProps = state => ({
    slideEditMode: state.slideEditMode,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            setEditSlideFalse
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(TeacherSaveEditsSlideButton);