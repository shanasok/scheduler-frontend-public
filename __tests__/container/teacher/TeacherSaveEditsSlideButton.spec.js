import React from 'react';
import TeacherSaveEditsSlideButton, {saveAndCloseSlideEditor} from '../../../src/container/teacher/TeacherSaveEditsSlideButton';
import renderer from 'react-test-renderer';
import configureMockStore from "redux-mock-store";
import {Provider as ReduxProvider} from "react-redux";
import {configure, mount} from "enzyme";
import {MemoryRouter} from "react-router";
import Adapter from "enzyme-adapter-react-16";
import {setEditSlideFalse} from "../../../src/redux/actions/editSlideAction";


configure({adapter: new Adapter()});

it('renders correctly when setEditSlide boolean is true', () => {

    const mockStore = configureMockStore()
    const mockState = {
        slideEditMode: true,
    }
    let store = mockStore(mockState);

    const tree = renderer
        .create( <ReduxProvider store={store}><TeacherSaveEditsSlideButton/></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children).toContain("Save Changes");
});

describe('testing saveAndCloseSlideEditor', () => {
    it('testing saveAndCloseSlideEditor calls reducer method that is passed in', () => {
        const mockFunction = jest.fn();
        saveAndCloseSlideEditor(mockFunction);
        expect(mockFunction.mock.calls.length).toBe(1);
    });

    it('testing onClick calls the function we pass in.', () => {
        const mockStore = configureMockStore()
        const mockFunction = jest.fn(setEditSlideFalse);

        const mockState = {
            slideEditMode: true,
            actions: { setEditSlideFalse: mockFunction },
        }

        let store = mockStore(mockState)

        let wrapper = mount(
            <MemoryRouter initialEntries={[ '/' ]}>
                <ReduxProvider store={store}>
                    <TeacherSaveEditsSlideButton />
                </ReduxProvider>
            </MemoryRouter>
        );

        wrapper.find('.saveEditsButton').at(0).simulate('click');
        expect(store.getActions()[0].type).toEqual('SET_EDIT_SLIDE_FALSE');
    });
});