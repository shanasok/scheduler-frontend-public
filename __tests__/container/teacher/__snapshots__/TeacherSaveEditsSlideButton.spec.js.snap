// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when setEditSlide boolean is true 1`] = `
<button
  className="saveEditsButton curvedButton"
  onClick={[Function]}
>
  Save Changes
</button>
`;
